src/main/java/winstone/AbstractSecuredConnectorFactory.java

```diff
@@ -122,10 +122,10 @@ protected void configureSsl( Map<String, String> args, Server server ) throws IO
     }
 
 
-    private static PrivateKey readPEMRSAPrivateKey(Reader reader) throws IOException, GeneralSecurityException {
+    private static PrivateKey readPEMRSAPrivateKey(File opensslKey) throws IOException, GeneralSecurityException {
         // TODO: should have more robust format error handling
         try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
-            try (BufferedReader r = new BufferedReader( reader )) {
+            try (BufferedReader r = Files.newBufferedReader(opensslKey.toPath(), StandardCharsets.US_ASCII)) {
```

Comment:
This is Base64-encoded per [RFC3447 - Appendix A.1.2](https://www.rfc-editor.org/rfc/rfc3447#appendix-A.1.2). See also the Base64 decode call a few lines below. Since this should not contain any special characters, ASCII should suffice.